src/bridge/cockpitcpusamples.c

```diff
@@ -112,3 +113,84 @@ cockpit_cpu_samples (CockpitSamples *samples)
   g_strfreev (lines);
   g_free (contents);
 }
+
+void
+cockpit_cpu_temperature (CockpitSamples *samples)
+{
+  gchar *name_content = NULL;
+  gchar *temp_content = NULL;
+  gchar *label_content = NULL;
+  GError *error = NULL;
+  gsize len;
+  gchar path[CPU_STR_MAXLEN];
+  gchar label[CPU_STR_MAXLEN];
+  gchar name[CPU_STR_MAXLEN];
+  guint32 temperature;
+
+  for (int i = 0; error == NULL; i++)
```

Comment:
I feel like maybe `for` is the wrong choice of loop here, and certainly using the condition like this makes things strange.  Some kind of an explicitly infinite-looping construct with well-placed `break;` on errors would probably read a lot nicer.